takwimu/takwimu_ui/src/components/Navigation/index.js

```diff
@@ -72,10 +76,17 @@ class Navigation extends React.Component {
   }
 
   toggleDrawer(drawer) {
+    const { openDrawer, isMobileDrawerOpen } = this.state;
+    const newOpenDrawer = openDrawer !== drawer ? drawer : null;
+
     return () => {
-      this.setState(prevState => ({
-        openDrawer: prevState.openDrawer !== drawer ? drawer : null
-      }));
+      const { width } = this.props;
+      this.setState({
+        isMobileDrawerOpen: isWidthUp('md', width)
+          ? false
+          : newOpenDrawer !== null || isMobileDrawerOpen,
+        openDrawer: openDrawer !== drawer ? drawer : null
```

Comment:
Isn't this line exactly like `newOpenDrawer` declaration above? Why not `openDrawer: newOpenDrawer`?

Reply:
good catch